iceoryx_posh/include/iceoryx_posh/internal/popo/building_blocks/typed_unique_id.hpp

```diff
@@ -11,33 +11,46 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-#ifndef IOX_UTILS_CXX_TYPED_UNIQUE_ID_HPP
-#define IOX_UTILS_CXX_TYPED_UNIQUE_ID_HPP
+#ifndef IOX_POSH_POPO_BUILDING_BLOCKS_TYPED_UNIQUE_ID_HPP
+#define IOX_POSH_POPO_BUILDING_BLOCKS_TYPED_UNIQUE_ID_HPP
```

Comment:
why now in the popo bulding blocks? I thought that this is a generic unique ID helper that we can use for other things too. I guess because of the RouDi ID that is in now. But than this becomes a very specific thing and is more ore less a Port ID and no more a generic UniqueID. Is it still worth the effort?

Reply:
The newtype pattern is still worth the effort since we would like to use it for instance to distinct:
 - Service, Instance, Event string by creating own types which would allow us to fetch even more errors at compile time
 - Service, Instance, Event id integers

The UniqueId is now part of the building blocks since it is now consisting of two parts:
- 16 bit RouDi id
- 48 bit Unique id